src/snapshot.rs

```diff
@@ -52,6 +53,26 @@ pub struct StartupData {
   raw_size: int,
 }
 
+impl From<Box<[u8]>> for StartupData {
+  fn from(slice: Box<[u8]>) -> Self {
+    let mut data: Vec<u8> = vec![];
+    data.copy_from_slice(&slice);
+    StartupData {
+      data: data.as_ptr() as *const char,
+      raw_size: data.len() as int,
+    }
+  }
+}
+
+impl From<&[u8]> for StartupData {
+  fn from(slice: &[u8]) -> Self {
+    StartupData {
+      data: slice.as_ptr() as *const char,
```

Comment:
You can't put rust data in StartupData - doesn't matter if it's owned or borrowed. This is because StartupData has a C++ destructor that calls `delete[] this->data;`.